samza-kafka/src/main/java/org/apache/samza/config/KafkaConsumerConfig.java

```diff
@@ -165,44 +168,64 @@ public static String createClientId(String prefix, Config config) {
     final String KAFKA_OFFSET_LATEST = "latest";
     final String KAFKA_OFFSET_EARLIEST = "earliest";
     final String KAFKA_OFFSET_NONE = "none";
 
-    if (autoOffsetReset == null) {
-      return KAFKA_OFFSET_LATEST; // return default
-    }
+    // if the value for KafkaConsumer is set - use it.
+    if (!StringUtils.isBlank(autoOffsetReset)) {
+      if (autoOffsetReset.equals(KAFKA_OFFSET_EARLIEST) || autoOffsetReset.equals(KAFKA_OFFSET_LATEST)
+          || autoOffsetReset.equals(KAFKA_OFFSET_NONE)) {
+        return autoOffsetReset;
+      }
+      // translate old kafka consumer values into new ones (SAMZA-1987 top remove it)
+      String newAutoOffsetReset = null;
+      switch (autoOffsetReset) {
+        case "largest":
+          newAutoOffsetReset = KAFKA_OFFSET_LATEST;
+          LOG.warn("Using old (deprecated) value for kafka consumer config auto.offset.reset = {}. The right value should be {}", autoOffsetReset, KAFKA_OFFSET_LATEST);
+          break;
+        case "smallest":
+          newAutoOffsetReset = KAFKA_OFFSET_EARLIEST;
+          LOG.warn("Using old (deprecated) value for kafka consumer config auto.offset.reset = {}. The right value should be {}", autoOffsetReset, KAFKA_OFFSET_EARLIEST);
+          break;
+        default:
+          throw new SamzaException("Using invalid value for kafka consumer config auto.offset.reset " + autoOffsetReset + ". See samza config for the correct values.");
+      }
 
-    // accept kafka values directly
-    if (autoOffsetReset.equals(KAFKA_OFFSET_EARLIEST) || autoOffsetReset.equals(KAFKA_OFFSET_LATEST)
-        || autoOffsetReset.equals(KAFKA_OFFSET_NONE)) {
-      return autoOffsetReset;
+      LOG.info("AutoOffsetReset value converted from {} to {}", autoOffsetReset, newAutoOffsetReset);
+      return newAutoOffsetReset;
     }
 
-    String newAutoOffsetReset;
-    switch (autoOffsetReset) {
-      case SAMZA_OFFSET_LARGEST:
-        newAutoOffsetReset = KAFKA_OFFSET_LATEST;
-        break;
-      case SAMZA_OFFSET_SMALLEST:
-        newAutoOffsetReset = KAFKA_OFFSET_EARLIEST;
-        break;
-      default:
-        newAutoOffsetReset = KAFKA_OFFSET_LATEST;
+    // in case kafka consumer configs are not provided we should match them to Samza's ones.
+    String newAutoOffsetReset = KAFKA_OFFSET_LATEST;
+    if (!StringUtils.isBlank(samzaOffsetDefault)) {
+      switch (samzaOffsetDefault) {
+        case JavaSystemConfig.SAMZA_SYSTEM_OFFSET_UPCOMING:
+          newAutoOffsetReset = KAFKA_OFFSET_LATEST;
+          break;
+        case JavaSystemConfig.SAMZA_SYSTEM_OFFSET_OLDEST:
+          newAutoOffsetReset = KAFKA_OFFSET_EARLIEST;
+          break;
+        default:
+          throw new SamzaException("Using invalid value for samza default offset config " + autoOffsetReset + ". See samza config for the correct values");
+      }
+      LOG.info("AutoOffsetReset value for converted from samza's {} to {}", samzaOffsetDefault, newAutoOffsetReset);
```

Comment:
s/AutoOffsetReset/Auto offset reset.

Should log system name in logs in this method.